src/Magento/FunctionalTestingFramework/Console/StaticChecksCommand.php

```diff
@@ -164,4 +160,43 @@ private function validateInput(InputInterface $input)
                 );
         }
     }
+
+    /**
+     * Validates that all passed in static-check names match an existing static check
+     * @param string[] $requiredChecksNames
+     * @return void
+     */
+    private function validateTestNames($requiredChecksNames)
+    {
+        $invalidCheckNames = [];
+        for ($index = 0; $index < count($requiredChecksNames); $index++) {
+            if (in_array($requiredChecksNames[$index], array_keys($this->allStaticCheckObjects))) {
+                $this->staticCheckObjects[$requiredChecksNames[$index]] =
+                    $this->allStaticCheckObjects[$requiredChecksNames[$index]];
+            } else {
+                $invalidCheckNames[] = $requiredChecksNames[$index];
+            }
+        }
+
+        if (!empty($invalidCheckNames)) {
+            throw new InvalidArgumentException(
+                'Invalid static check script(s): ' . implode(', ', $invalidCheckNames) . '.'
+            );
+        }
+    }
+
+    /**
+     * Parses and sets local ruleSet. If not found, simply returns and lets script continue.
+     * @return void;
+     */
+    private function parseRulesetJson()
+    {
+        $pathToRuleset = TESTS_BP . DIRECTORY_SEPARATOR . "staticRuleset.json";
```

Comment:
should the file `staticRuleset.json` even be added to MFTF? Because, again updates to this file will need a new MFTF version. Reading from this path works for both standalone and embedded -
```
const DEV_TESTS_DIR = 'dev/tests/acceptance/';        
$devTestsPath = FilePathFormatter::format(MAGENTO_BP) . self::DEV_TESTS_DIR;
$pathToRuleset = $devTestsPath . DIRECTORY_SEPARATOR . "staticRuleset.json";
```

Reply:
See other comment, TESTS_BP is set as following:
* If we detect we are attached to Magento (either by composer or MAGENTO_BP), it's `dev/tests/acceptance`
* If we are not attached, it's just `dev/` to point towards MFTF standalone `dev directory`

Reply:
in standalone mode, with latest code, it tries to read `staticRuleSet.json` from mftf's `\dev`, doesn't find it, and executes all static checks instead. Shouldn't standalone also read from attached magento's `dev/tests/acceptance`?

However in case mftf is not attached at all i.e MAGENTO_BP is not set in `.env`, it again tries to read file from mftf's `\dev`, doesn't find it and executes all static checks, which seems correct.

Reply:
Also found another bug, though unrelated to this PR -
if MAGENTO_BP is not set in `.env` (detached mode) , deprecatedEntityUsage throws php errors. 
Simple fix would be to add an error message like `[ERROR] DEPRECATED ENTITY USAGE CHECK ABORTED: MFTF must be attached or pointing to Magento codebase.`

Reply:
I think I understand better now, let me try and reproduce scenario 1.